### ISSUE TITLE
Web Components are removed from phrasing-content-only blocks

### ISSUE THREAD

**Issue by jshcrowthe** [kangax/html-minifier#557]
(Copy and paste from: https://github.com/kangax/html-minifier/pull/541#issuecomment-197554845)

Great work with the spec compliance here! I am using html-minifier (via gulp) for a while now and it has been great. However, in an application using web components, this addition broke me.

Web Components have the unique nature that they can be put in any of the available content categories. In order for this to work with those using web components, tags that are not, by default, part of the HTML spec (and therefore recognized as phrasing, flow, or another type) should be ignored.

I added the following test to https://github.com/kangax/html-minifier/blob/gh-pages/tests/minifier.js#L1191 and it failed:

```
  test('phrasing content with Web Components', function() {
    input = '<span><phrasing-element></phrasing-element></span>';
    output = '<span><phrasing-element></phrasing-element></span>';
    equal(minify(input, { html5: true }), output);
  });
```

**Comment by pavax** [comment]
+1

**Comment by mathieu-clement** [comment]
+1

**Comment by mathieu-clement** [comment]
If I look at the code in htmlparser.js I can see that tags such as h3 are listed as phrasing content. This however is not what the HTML specification says:
https://html.spec.whatwg.org/multipage/dom.html#phrasing-content-2

You can see that h3 is not in this list. It is listed as "flow content".

**Comment by alexlamsl** [comment]
[No they are not](https://github.com/kangax/html-minifier/blob/gh-pages/src/htmlparser.js#L91).

I think you may be confusing the tags that can contain [phrasing content _only_](https://html.spec.whatwg.org/multipage/semantics.html#the-h1,-h2,-h3,-h4,-h5,-and-h6-elements).

**Comment by alexlamsl** [comment]
Fixed by #558

**Comment by tiktaktok** [comment]
I'm confused, because h1-h6 are listed here:
https://html.spec.whatwg.org/multipage/dom.html#flow-content
